Px.Utils/Validation/SyntaxValidation/SyntaxValidation.cs

```diff
@@ -0,0 +1,341 @@
+            int character = 0;
+            bool isProcessingString = false;
+
+            using StreamReader reader = new(stream, encoding);
+            List<ValidationEntry> entries = [];
+            StringBuilder entryBuilder = new();
+            char[] buffer = new char[bufferSize];
+            int read = 0;
+            do
+            {
+                read = await reader.ReadAsync(buffer.AsMemory(), cancellationToken);
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    if (IsEndOfMetadataSection(buffer[i], syntaxConf, entryBuilder, isProcessingString))
+                    {
+                        break;
+                    }
+                    UpdateLineAndCharacter(buffer[i], syntaxConf, ref line, ref character, ref isProcessingString);
+                    if (IsEndOfEntry(buffer[i], syntaxConf, isProcessingString))
+                    {
+                        string stringEntry = entryBuilder.ToString();
+                        entries.Add(new ValidationEntry(line, character, filename, stringEntry, entries.Count));
+                        entryBuilder.Clear();
+                    }
+                    else
+                    {
+                        entryBuilder.Append(buffer[i]);
+                    }
+                }
+                Array.Clear(buffer, 0, buffer.Length);
+            }
+            while (read > 0);
+
+            return entries;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsEndOfMetadataSection(char currentCharacter, PxFileSyntaxConf syntaxConf, StringBuilder entryBuilder, bool isProcessingString)
+        {
+            if (!isProcessingString && currentCharacter == syntaxConf.Symbols.KeywordSeparator)
+            {
+                string stringEntry = entryBuilder.ToString();
+                // When DATA keyword is reached, metadata parsing is complete
+                return SyntaxValidationUtilityMethods.CleanString(stringEntry, syntaxConf).Equals(syntaxConf.Tokens.KeyWords.Data);
+            }
+            return false;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void UpdateLineAndCharacter(char currentCharacter, PxFileSyntaxConf syntaxConf, ref int line, ref int character, ref bool isProcessingString)
+        {
+            if (currentCharacter == syntaxConf.Symbols.Linebreak)
+            {
+                line++;
+                character = 0;
+            }
+            else
+            {
+                if (currentCharacter == syntaxConf.Symbols.Key.StringDelimeter)
+                {
+                    isProcessingString = !isProcessingString;
+                }
+                character++;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsEndOfEntry(char currentCharacter, PxFileSyntaxConf syntaxConf, bool isProcessingString)
```

Comment:
Call to this function can be replaced with !isProcessingString && currentCharacter == syntaxConf.Symbols.EntrySeparator